### ISSUE TITLE
Problem installing

### ISSUE THREAD

**Issue by buzzbo** [randomchars/pushbullet.py#36]
Trying to install with pip install pushbullet.py and get errors. I have python-magic and requests. Running on Raspberry pi (raspbian).

    /usr/bin/pip run on Wed Mar 11 19:36:16 2015
    Downloading/unpacking pushbullet.py
    
      Getting page http://pypi.python.org/simple/pushbullet.py
      URLs to search for versions for pushbullet.py:
      * http://pypi.python.org/simple/pushbullet.py/
      Getting page http://pypi.python.org/simple/pushbullet.py/
      Analyzing links from page https://pypi.python.org/simple/pushbullet.py/
        Skipping link https://pypi.python.org/packages/2.7/p/pushbullet.py/pushbullet.py-0.4.1-py2.py3-none-any.whl#md5=5fdbecf2195ce5009e762c2a91fa6be3 (from https://pypi.python.org/simple/pushbullet.py/); unknown archive format: .whl
        Skipping link https://pypi.python.org/packages/3.4/p/pushbullet.py/pushbullet.py-0.7.0-py2.py3-none-any.whl#md5=c3a7496b32ff8af14af8b0db8b787606 (from https://pypi.python.org/simple/pushbullet.py/); unknown archive format: .whl
        Skipping link https://pypi.python.org/packages/3.4/p/pushbullet.py/pushbullet.py-0.7.1-py2.py3-none-any.whl#md5=a2e4a6b2f64715c0bb264d951ecb1464 (from https://pypi.python.org/simple/pushbullet.py/); unknown archive format: .whl
        Skipping link https://pypi.python.org/packages/3.4/p/pushbullet.py/pushbullet.py-0.8.0-py2.py3-none-any.whl#md5=34c2642c1b57139af44363709a56bc74 (from https://pypi.python.org/simple/pushbullet.py/); unknown archive format: .whl
        Skipping link https://pypi.python.org/packages/3.4/p/pushbullet.py/pushbullet.py-0.8.1-py2.py3-none-any.whl#md5=8f25da4f267d3bc2ae38ec5b2fa37706 (from https://pypi.python.org/simple/pushbullet.py/); unknown archive format: .whl
        Found link https://pypi.python.org/packages/source/p/pushbullet.py/pushbullet.py-0.5.0.tar.gz#md5=36c83ba5f7d5208bb86c00eba633f921 (from https://pypi.python.org/simple/pushbullet.py/), version: 0.5.0
        Found link https://pypi.python.org/packages/source/p/pushbullet.py/pushbullet.py-0.7.0.tar.gz#md5=58f603d1c2fd13dec443b48130bac6bf (from https://pypi.python.org/simple/pushbullet.py/), version: 0.7.0
        Found link https://pypi.python.org/packages/source/p/pushbullet.py/pushbullet.py-0.7.1.tar.gz#md5=fe1cf277f3a91fce5692a5cf923b32c5 (from https://pypi.python.org/simple/pushbullet.py/), version: 0.7.1
        Found link https://pypi.python.org/packages/source/p/pushbullet.py/pushbullet.py-0.8.0.tar.gz#md5=09beddda880c336c2fd693bf147d5ca4 (from https://pypi.python.org/simple/pushbullet.py/), version: 0.8.0
        Found link https://pypi.python.org/packages/source/p/pushbullet.py/pushbullet.py-0.8.1.tar.gz#md5=1e8eccd95062df199830d92a29c57bdf (from https://pypi.python.org/simple/pushbullet.py/), version: 0.8.1
      Using version 0.8.1 (newest of versions: 0.8.1, 0.8.0, 0.7.1, 0.7.0, 0.5.0)
      Downloading pushbullet.py-0.8.1.tar.gz
    
      Downloading from URL https://pypi.python.org/packages/source/p/pushbullet.py/pushbullet.py-0.8.1.tar.gz#md5=1e8eccd95062df199830d92a29c57bdf (from https://pypi.python.org/simple/pushbullet.py/)
      Running setup.py egg_info for package pushbullet.py
    
        Traceback (most recent call last):
    
          File "<string>", line 14, in <module>
    
          File "/root/build/pushbullet.py/setup.py", line 15, in <module>
    
            install_reqs = parse_requirements("requirements.txt",  session=uuid.uuid1())
    
        TypeError: parse_requirements() got an unexpected keyword argument 'session'
    
        Complete output from command python setup.py egg_info:
    
        Traceback (most recent call last):
    
      File "<string>", line 14, in <module>
    
      File "/root/build/pushbullet.py/setup.py", line 15, in <module>
    
        install_reqs = parse_requirements("requirements.txt",  session=uuid.uuid1())
    
    TypeError: parse_requirements() got an unexpected keyword argument 'session'
    
    ----------------------------------------
    
    Command python setup.py egg_info failed with error code 1 in /root/build/pushbullet.py
    
    Exception information:
    Traceback (most recent call last):
      File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 104, in main
        status = self.run(options, args)
      File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 245, in run
        requirement_set.prepare_files(finder, force_root_egg_info=self.bundle, bundle=self.bundle)
      File "/usr/lib/python2.7/dist-packages/pip/req.py", line 1009, in prepare_files
        req_to_install.run_egg_info()
      File "/usr/lib/python2.7/dist-packages/pip/req.py", line 225, in run_egg_info
        command_desc='python setup.py egg_info')
      File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 256, in call_subprocess
        % (command_desc, proc.returncode, cwd))
    InstallationError: Command python setup.py egg_info failed with error code 1 in /root/build/pushbullet.py

**Comment by markcs** [comment]
same issue

**Comment by christian512** [comment]
Got the same Problem

**Comment by christian512** [comment]
I had the same issue. So I got Python3.4 to my Raspberry Pi and tried to install with pip3.4. But I now get a PermissionError.


```
pi@raspberrypi ~/pushbullet.py $ pip3.4 install -r requirements.txt
Downloading/unpacking requests>=1.0.0 (from -r requirements.txt (line 1))
  Downloading requests-2.6.0-py2.py3-none-any.whl (469kB): 469kB downloaded
Downloading/unpacking python-magic (from -r requirements.txt (line 2))
  Downloading python-magic-0.4.6.tar.gz
  Running setup.py (path:/tmp/pip_build_pi/python-magic/setup.py) egg_info for package python-magic
    
Downloading/unpacking websocket-client (from -r requirements.txt (line 3))
  Downloading websocket_client-0.26.0.tar.gz (189kB): 189kB downloaded
  Running setup.py (path:/tmp/pip_build_pi/websocket-client/setup.py) egg_info for package websocket-client
    
Downloading/unpacking six (from websocket-client->-r requirements.txt (line 3))
  Downloading six-1.9.0-py2.py3-none-any.whl
Installing collected packages: requests, python-magic, websocket-client, six
Cleaning up...
Exception:
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.4/site-packages/pip/basecommand.py", line 122, in main
    status = self.run(options, args)
  File "/home/pi/.local/lib/python3.4/site-packages/pip/commands/install.py", line 283, in run
    requirement_set.install(install_options, global_options, root=options.root_path)
  File "/home/pi/.local/lib/python3.4/site-packages/pip/req.py", line 1435, in install
    requirement.install(install_options, global_options, *args, **kwargs)
  File "/home/pi/.local/lib/python3.4/site-packages/pip/req.py", line 671, in install
    self.move_wheel_files(self.source_dir, root=root)
  File "/home/pi/.local/lib/python3.4/site-packages/pip/req.py", line 901, in move_wheel_files
    pycompile=self.pycompile,
  File "/home/pi/.local/lib/python3.4/site-packages/pip/wheel.py", line 215, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/home/pi/.local/lib/python3.4/site-packages/pip/wheel.py", line 205, in clobber
    os.makedirs(destdir)
  File "/home/pi/.local/lib/python3.4/os.py", line 237, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/home/pi/.local/lib/python3.4/site-packages/requests-2.6.0.dist-info'

Storing debug log for failure in /home/pi/.pip/pip.log
```

**Comment by buzzbo** [comment]
I think my problem was that I was using the raspbian package pip which is evidently outdated. I uninstalled and reinstalled a newer version using the instructions below.

https://pip.pypa.io/en/latest/installing.html

Christian, I think you need to execute yours as superuser (sudo pip...)

**Comment by christian512** [comment]
However I can not run pip as sudo. I will try it with the newer version of pip.

**Comment by rmjspa** [comment]
I have the same problem as above. I tried also the newer version of pip. Also, when I clone and run install I get always the same result.

pi@Master ~ $ sudo pip install pushbullet.py
Downloading/unpacking pushbullet.py
  Downloading pushbullet.py-0.8.1.tar.gz
  Running setup.py egg_info for package pushbullet.py
    Traceback (most recent call last):
      File "<string>", line 14, in <module>
      File "/home/pi/build/pushbullet.py/setup.py", line 15, in <module>
        install_reqs = parse_requirements("requirements.txt",  session=uuid.uuid1())
    TypeError: parse_requirements() got an unexpected keyword argument 'session'
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 14, in <module>

  File "/home/pi/build/pushbullet.py/setup.py", line 15, in <module>

    install_reqs = parse_requirements("requirements.txt",  session=uuid.uuid1())

TypeError: parse_requirements() got an unexpected keyword argument 'session'

----------------------------------------
Command python setup.py egg_info failed with error code 1 in /home/pi/build/pushbullet.py
Storing complete log in /root/.pip/pip.log

Three weeks ago I got it to run some how, but after a brand new installation i try since two days without luck.
Has anyone  a solution for that problem?
